src/gtest/test_pow.cpp

```diff
@@ -68,3 +68,44 @@ TEST(PoW, DifficultyAveraging) {
                                         params),
               GetNextWorkRequired(&blocks[lastBlk], nullptr, params));
 }
+
+TEST(PoW, MinDifficultyRules) {
+    SelectParams(CBaseChainParams::TESTNET);
```

Comment:
How does this test work given that min-difficulty blocks aren't allowed on testnet until after block 299187?

Reply:
Whoops, yep I forgot to update this test when adding the restriction. Pushed a fix.

Reply:
The fix uses regtest; I would mildly prefer it to use testnet and fake the height somehow. Using regtest will do if this is too hard; I don't want to delay the release too much longer.

Reply:
Modified the fix to use testnet instead. Opened #3565 to test the pre-activation part of the rule, which would require more extensive test changes.